infinity-app/app.R

```diff
@@ -301,38 +365,69 @@ server <- function(input, output, session) {
   })
   
   
-  # Observer to update genus input based on the selected place
-  observeEvent(input$place, {
-        updateSelectizeInput(
-          session,
-          "taxa_genus",
-          selected = "Eucalyptus",
-          choices = update_genus_choices(input$place),
-          server = FALSE)
-        updateSelectizeInput(
-          session,
-          "taxa_family",
-          choices = update_family_choices(input$place),
-          selected = "Myrtaceae",
-          server = FALSE
-        )
+  ala_data <- reactive({
+    long_buffer <- 0.578 #50 km approx
+    lat_buffer <- 0.45 #50 km approx
+    place_polygon <- selected_polygon()
+    open_dataset(paste0("data/", input$ala_path)) |> 
+      filter(Lat< st_bbox(place_polygon)$ymax + lat_buffer & 
+             Lat > st_bbox(place_polygon)$ymin - lat_buffer&
+             Long < st_bbox(place_polygon)$xmax + long_buffer & 
+             Long > st_bbox(place_polygon)$xmin - long_buffer) |> 
+      collect() |> data.table() 
+  })
+  
+  # A reactive to combine your two inputs
+  combined_input <- reactive({
+    list(place = input$place, ala_path = input$ala_path)
+  })
+  
+  # Observe changes in the combined input
+  observeEvent(combined_input(), {
+    
+    # Directly get the data from the ala_data reactive
+    current_ala <- ala_data()
+    
+    most_common_genus <- names(sort(table(current_ala$Genus), decreasing = TRUE)[1])
+    most_common_family <- names(sort(table(current_ala$Family), decreasing = TRUE)[1])
+    
+    updateSelectizeInput(
+      session,
+      "taxa_genus",
+      selected = most_common_genus,
+      choices = update_genus_choices(input$place),
+      server = FALSE)
+    
+    updateSelectizeInput(
+      session,
+      "taxa_family",
+      choices = update_family_choices(input$place),
+      selected = most_common_family,
+      server = FALSE
+    )
   })
+
    
   # Reactive expression to summarize and filter data
-  filtered_data<- reactive({ 
-    result<-intersect_data()
+  filtered_data <- reactive({ 
+    result <- intersect_data()
     if (nrow(result) == 0) {
       return(data.table(Species = character(0), `Voucher Type` = character(0), `Most recent obs.` = character(0), N = integer(0), Long = numeric(0), Lat = numeric(0), `Voucher location` = character(0), `Observed by` = character(0), Native = character(0)))
     }
+    
+    # Sort the data by 'in target area' and 'Collection Date'
+    result <- result[order(`in target area` == "in target", -`Collection Date`)]
+    
+    
     result <- result[, .(
+      `in target area` = `in target area`[1],
```

Comment:
Nitpicking, everything in table is capitalised :D

Reply:
fixed